src/js/jsx/search/SearchBar.jsx

```diff
@@ -91,7 +93,8 @@ define(function (require, exports, module) {
             this.setState({
```

Comment:
@iwehrman I made it a StoreWatchMixin, which will be useful for picking up new providers, but there wasn't an apparent way to remove the call to update the store from component did mount. If it is called any earlier, then the options haven't loaded yet in the document/application stores, so we just get empty lists of layers and documents.

The only way around it that I can think of would be to add listeners to actions that indicate when the layers/documents are ready, but those would have to be specific based on the providers, so we'd have to modify the search store whenever we wanted to add a new type of provider.